internal/cloud/realm/client.go

```diff
@@ -45,13 +47,13 @@ func NewClient(baseURL string) Client {
 }
 
 // NewAuthClient creates a new Realm client with a session used for Authorization
-func NewAuthClient(baseURL string, session Session) Client {
-	return &client{baseURL, session}
+func NewAuthClient(baseURL string, profile *profile.Profile) Client {
+	return &client{baseURL, profile}
 }
 
 type client struct {
 	baseURL string
-	session Session
+	profile *profile.Profile
```

Comment:
since we get the baseURL from the client, we could just have profile in the client struct. It does raise the question of how necessary it is to have a client that wraps around a single value. But I do like the notion of having the doJson, do, and other methods on client rather than Profile